### PR TITLE
Refactor identity helper

### PR DESCRIPTION
**Pull request by gsteel**
### Description

The identity helper imports laminas auth which makes it an undeclared runtime dependency.

Because only this helper (and its factory) uses the laminas auth package, the helper has been altered to check for the methods `hasIdentity` and `getIdentity`.

Submitting the pull without test alterations in the first commit so that CI can pass before the tests are changed

### CODE REVIEW

**Comment by Xerkus** [comment]
Laminas Authentication is a soft dependency. Interface does not have to be declared to be used for checks as in the existing code. No autoload attempt is happening when instanceof check is used.

Removing interface reduces type safety for no gain.

File: src/Helper/Identity.php [hunk 1]
```diff
@@ -4,57 +4,70 @@
...
-use Laminas\View\Exception;
+use function method_exists;
 
 /**
  * View helper plugin to fetch the authenticated identity.
  */
 class Identity extends AbstractHelper
 {
-    /**
-     * AuthenticationService instance
```
Comment (Ocramius) [inline src/Helper/Identity.php:16]: IMO to be reverted: this code does not initialize an `AuthenticationSetvice`, but accepts an already existing one.

A type level declaration is preferable to going back to `object`, even if it leads to a soft dependency.

Stubbing it out could suffice, but the removal of it from `composer.json` potentially leads to conflicts with new major releases of the authentication component. Perhaps replacing `require` with `conflict` would suffice here.

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by gsteel** [comment]
@Xerkus and @Ocramius - Reverted the duck typing so it's just a general clean up and deprecation of runtime get/set of the auth service.

File: src/Helper/Service/IdentityFactory.php [hunk 2]
```diff
@@ -41,23 +38,30 @@ public function createService(ServiceLocatorInterface $serviceLocator, $rName =
...
-     */
-    private function discoverAuthenticationService(ContainerInterface $container)
+    private function discoverAuthenticationService(ContainerInterface $container): ?AuthenticationServiceInterface
     {
-        if ($container->has(AuthenticationService::class)) {
-            return $container->get(AuthenticationService::class);
-        }
+        // phpcs:disable WebimpressCodingStandard.Formatting.StringClassReference
+        $search = [
+            AuthenticationService::class,
```
Comment (Ocramius) [inline src/Helper/Service/IdentityFactory.php:45]: IMO start from the `*Interface`, then fallback to concrete classes :thinking:

Comment (Ocramius) [reply src/Helper/Service/IdentityFactory.php:45]: Hmm, I see this was the previous behavior (checking concrete implementation first) - nvm :|

**Review by Ocramius — APPROVED** [review]
LGTM :+1:

**Comment by Ocramius** [comment]
Thanks @gsteel!

**Comment by gsteel** [comment]
Thanks too to you @Ocramius for your help and reviews of all this stuff 😃

**Pull request merged**